db/db.go

```diff
@@ -194,13 +194,31 @@ func (l *Loader) GetIdentifier() string {
 	return fmt.Sprintf("%s/%s", l.database, l.schema)
 }
 
-func (l *Loader) GetAvailableTablesInSchema() string {
+
+func (l *Loader) GetAvailableTablesInSchemaList() []string {
 	tables := make([]string, len(l.tables))
 	i := 0
 	for table := range l.tables {
 		tables[i] = table
 		i++
 	}
+	return tables
+}
+
+func (l *Loader) GetColumnsForTable(name string) []string {
+	// we can't use fixed size because, for some reason, some of them may be empty
+	columns := []string{}
```

Comment:
Small trick for pre-allocation when you know empty column is something very rare:

```suggestion
	columns := make([]string, 0, len(l.tables[name].columnsByName))
```

This pre-allocate the slice but the count is empty, `append(columns, ...)` will make no allocation in those cases.

Reply:
For some reason, when I tried to use this, I got some empty columns but I may not be testing correctly.

Reply:
Really need the first 0 which is the len of the slice, specifying `make([]string, len(l.tables[name].columnsByName))` would append a new element after the N empty one pre-allocated.

Reply:
Oh, it makes sense now!